src/config.rs

```diff
@@ -482,10 +494,15 @@ impl Account {
             && self.token_uri == act_dump.token_uri
     }
 
-    pub fn redirect_uri(&self, http_port: u16) -> Result<Url, Box<dyn Error>> {
+    pub fn redirect_uri(&self, http_port: u16, https_port: u16) -> Result<Url, Box<dyn Error>> {
```

Comment:
This should probably take in `&Arc<AuthenticatorState>` and fish the ports out of there directly.

Reply:
Not sure whats the best way here. This would require to make `mod state` being public or `AuthenticatorState` to be reexported by `server/mod.rs`. So far I think that's ok. However, for the test to work, also Eventer, Notifier and Refresher need to be exported - which I think they shouldn't. Any idea on that (as its just for the test)?

Reply:
You're right! Best leave this one as-is then.